lib/auth0/api/authentication_endpoints.rb

```diff
@@ -502,6 +506,51 @@ def unlink_user(access_token, user_id)
         post('/unlink', request_params)
       end
 
+      # Validate an ID token (signature and expiration).
+      # @see https://auth0.com/docs/tokens/guides/validate-id-tokens
+      # @param id_token [string] The JWT to validate.
+      # @param algorithm [object] The expected signing algorithm.
+      #   Defaults to Auth0::Algorithm::RS256.jwks_url("https://YOUR_AUTH0_DOMAIN/.well-known/jwks.json").
+      # @param leeway [integer] The clock skew to accept when verifying date related claims in seconds.
+      #   Must be a non-negative value. Defaults to 60 seconds.
+      # @param nonce [string] The nonce value sent during authentication.
+      # @param max_age [integer] The max_age value sent during authentication.
+      #   Must be a non-negative value.
+      # @param issuer [string] The expected issuer claim value.
+      #   Defaults to "https://YOUR_AUTH0_DOMAIN/".
+      # @param audience [string] The expected audience claim value.
+      #   Defaults to your Auth0 Client ID.
+      # rubocop:disable Metrics/MethodLength, Metrics/AbcSize, Metrics/ParameterLists
+      def validate_id_token(id_token, algorithm: nil, leeway: 60, nonce: nil, max_age: nil, issuer: nil, audience: nil)
+        raise Auth0::InvalidParameter, 'ID token is required but missing' if id_token.to_s.empty?
+        raise Auth0::InvalidParameter, 'Must supply a valid leeway' unless leeway.is_a?(Integer) && leeway >= 0
+        raise Auth0::InvalidParameter, 'Must supply a valid nonce' unless nonce.nil? || !nonce.to_s.empty?
+        raise Auth0::InvalidParameter, 'Must supply a valid issuer' unless issuer.nil? || !issuer.to_s.empty?
+        raise Auth0::InvalidParameter, 'Must supply a valid audience' unless audience.nil? || !audience.to_s.empty?
+
+        unless max_age.nil? || (max_age.is_a?(Integer) && max_age >= 0)
+          raise Auth0::InvalidParameter, 'Must supply a valid max_age'
+        end
+
+        unless algorithm.nil? || algorithm.is_a?(Auth0::Algorithm::RS256) || algorithm.is_a?(Auth0::Algorithm::HS256)
+          raise Auth0::InvalidParameter, "Signature algorithm of \"#{algorithm}\" is not supported"
+        end
+
+        context = {
+          issuer: issuer || "https://#{@domain}/",
+          audience: audience || @client_id,
+          algorithm: algorithm || Auth0::Algorithm::RS256.jwks_url("https://#{@domain}/.well-known/jwks.json"),
+          leeway: leeway
+        }
+
+        context[:nonce] = nonce unless nonce.nil?
```

Comment:
it could be added anyway above in the { } definition and checked down in the claims validation section. There's no escape from that later check, so it seems best to simplify these two here.

Reply:
You're right, I'll change it.

Reply:
Changed in efdf3feec312ed9d2e8a91fa3ea9b625373033ec